run_tests.py

```diff
@@ -155,30 +156,36 @@ def prompt_user_to_run_tests(test_names):
         elif answer.upper()[0] == "Y":
             return
 
+def runTestProto(self):
+    """
+    Function to be used as basis for runTest function of ReportFailLoadTestsuiteTestCase.
+    We create a specific named test function to improve the test failure summary details.
+    """
+    self.fail(self.msg)
 
 class ReportFailLoadTestsuiteTestCase(unittest.TestCase):
     """
     Class to allow reporting of an error to run any tests.
 
     Args:
-        msg: Error message.
+        func_name:   Name of function to create for the test.
+        msg:         Error message explaining test failure.
 
     Returns:
         None
     """
-    def __init__(self, msg):
-        super(ReportFailLoadTestsuiteTestCase,self).__init__()
+    def __init__(self, func_name, msg):
+        setattr(self, func_name, types.MethodType(runTestProto, self))
+        super(ReportFailLoadTestsuiteTestCase,self).__init__(func_name)
```

Comment:
This is setting the method name to run to be the same as function that doesn't seem right. Also unit test says:

```    If it is necessary to override the __init__ method, the base class
    __init__ method must always be called. It is important that subclasses
    should not change the signature of their __init__ method, since instances
    of the classes are instantiated automatically by parts of the framework
    in order to be run.
```

I think we should not create out own code. Although maybe you have already had this argument.

Reply:
Acrtually probably ok but leave a coment saying this is only use din our ioc testing framework

Reply:
Might not have to do this at all if the function you add is called test_xxx

Reply:
The function name (failed module name) need to pass to super init, so I passed it by subclass init - is there any other way other than a class variable?